Pixel 9 Pro XL: Crashloop after installing release 2024103100

I installed release `2024103100` and after unlocking the phone for the first time, I got caught in a crashloop from the GrapheneOS bootscreen to the launcher. 

I got it resolved by forcefully powering off the device and restarting. So far everything's stable again.

The logs below are from the system crash dialog and they end right when it gets interesting. I can pull the full log if it helps.

```
type: crash
flags: dev options enabled

SystemUptimeMs: 24959
Process: system_server
Timestamp: 2024-11-02 08:15:11.215+0100
Build: google/komodo/komodo:15/AP3A.241005.015/2024103100:user/release-keys
Crash-Handler: com.android.internal.os.RuntimeInit$KillApplicationHandler
Loading-Progress: 1.0
Dropped-Count: 0

android.content.res.Resources$NotFoundException: File res/layout/alert_dialog_material.xml from xml type layout resource ID #0x1090031
	at android.content.res.ResourcesImpl.loadXmlResourceParser(ResourcesImpl.java:1341)
	at android.content.res.Resources.loadXmlResourceParser(Resources.java:2594)
	at android.content.res.Resources.loadXmlResourceParser(Resources.java:2570)
	at android.content.res.Resources.getLayout(Resources.java:1284)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:460)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:413)
	at com.android.internal.policy.PhoneWindow.setContentView(PhoneWindow.java:537)
	at com.android.internal.app.AlertController.installContent(AlertController.java:269)
	at android.app.AlertDialog.onCreate(AlertDialog.java:439)
	at android.app.ProgressDialog.onCreate(ProgressDialog.java:272)
	at android.app.Dialog.dispatchOnCreate(Dialog.java:436)
	at android.app.Dialog.show(Dialog.java:325)
	at com.android.server.power.ShutdownThread.showShutdownDialog(ShutdownThread.java:368)
	at com.android.server.power.ShutdownThread.beginShutdownSequence(ShutdownThread.java:406)
	at com.android.server.power.ShutdownThread.shutdownInner(ShutdownThread.java:222)
	at com.android.server.power.ShutdownThread.reboot(ShutdownThread.java:262)
	at com.android.server.power.PowerManagerService$2.run(PowerManagerService.java:4047)
	at android.os.Handler.handleCallback(Handler.java:959)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.os.HandlerThread.run(HandlerThread.java:85)
	at com.android.server.ServiceThread.run(ServiceThread.java:46)
	at com.android.server.UiThread.run(UiThread.java:45)
Caused by: java.io.FileNotFoundException: res/layout/alert_dialog_material.xml
	at android.content.res.AssetManager.nativeOpenXmlAsset(Native Method)
	at android.content.res.AssetManager.openXmlBlockAsset(AssetManager.java:1232)
	at android.content.res.ResourcesImpl.loadXmlResourceParser(ResourcesImpl.java:1325)
	... 23 more
```

--- a/services/core/java/com/android/server/power/ShutdownThread.java
+++ b/services/core/java/com/android/server/power/ShutdownThread.java
@@ -403,7 +403,13 @@ public final class ShutdownThread extends Thread {
             sIsStarted = true;
         }
 
-        sInstance.mProgressDialog = showShutdownDialog(context);
+        try {
+            sInstance.mProgressDialog = showShutdownDialog(context);
+        } catch (android.content.res.Resources.NotFoundException e) {
+            // mProgressDialog is expected to be nullable, no special handling is neeeded for this case
+            Slog.e(TAG, "unable to show shutdown dialog", e);
+        }
+
         sInstance.mContext = context;
         sInstance.mPowerManager = (PowerManager)context.getSystemService(Context.POWER_SERVICE);
 

